pkg/infra/fs/copy_test.go

```diff
@@ -78,12 +78,24 @@ func TestCopyFile_NonExistentDestDir(t *testing.T) {
 func TestCopyRecursive_NonExistentDest(t *testing.T) {
 	src := t.TempDir()
 
-	err := os.MkdirAll(filepath.Join(src, "data"), 0750)
+	err := os.MkdirAll(filepath.Join(src, "data", "nested"), 0750)
 	require.NoError(t, err)
 	// nolint:gosec
 	err = os.WriteFile(filepath.Join(src, "data", "file.txt"), []byte("Test"), 0644)
 	require.NoError(t, err)
 
+	linkPath := filepath.Join(src, "data", "nested", "link.txt")
+
+	err = os.Symlink(filepath.Join(src, "data", "file.txt"), linkPath)
+	require.NoError(t, err)
+
+	lstat, err := os.Lstat(linkPath)
+	require.NoError(t, err)
+	require.Equal(t, os.ModeSymlink, lstat.Mode()&os.ModeType)
+	stat, err := os.Stat(linkPath)
+	require.NoError(t, err)
+	require.NotEqual(t, os.ModeSymlink, stat.Mode()&os.ModeType)
```

Comment:
@marefr i added this test as an example of the issue. `CopyRecursive` is using `os.Stat`  instead of `os.LStat` to check for `os.ModeSymlink`